assignments/hw4/diffs.py

```diff
@@ -0,0 +1,126 @@
+"""Module implements difference with mean of response.
+
+Reproduces formula and table given in class
+"""
+
+import sys
+
+import data
+import features
+import pandas as pd
+
+
+def calc_diff_with_mean_table_cont(
+    df: pd.DataFrame,
+    predictor: str,
+    response: str,
+    n: int = 10,
+) -> pd.DataFrame:
+    bin_df = pd.DataFrame()
+    ps = df[predictor]
+    rs = df[response]
+    pred_df = ps.to_frame()
+    pred_df[response] = rs
+    pop_mean = rs.mean()
+    # Based on lecture notes
+    # Bin candidate predictor variable
+    intervals = pd.qcut(ps, n)
+    pred_df["LowerBin"] = pd.Series([i.left for i in intervals])
+    pred_df["UpperBin"] = pd.Series([i.right for i in intervals])
```

Comment:
I've never seen .left and .right used before. I'll have to keep that in mind next time I need to do something similar! :+1: